src/System.Management.Automation/engine/parser/Compiler.cs

```diff
@@ -5500,16 +5502,28 @@ public object VisitArrayExpression(ArrayExpressionAst arrayExpressionAst)
             }
             values = values ?? CaptureAstResults(subExpr, CaptureAstContext.Enumerable);
 
-            if (values.Type.IsArray)
+            if (values.Type == typeof(object[]) || values.Type == typeof(List<object>))
```

Comment:
Replace the cases of `values.Type.IsArray` and `values.Type == typeof(List<object>)` with `if (pureExprAst is ArrayLiteralAst)`. So the former cases will be handled by the dynamic site.